client/src/User.re

```diff
@@ -3,44 +3,115 @@ open Utils;
         date: updated_at
       }
     }
-  |}
+    |}
 ];
 
 module GetNotesComponent = ReasonApollo.CreateQuery(GetNotes);
 
-let component = ReasonReact.statelessComponent("User");
+let getNotesQuery = (~userName, ~offset=0, ()) =>
+  GetNotes.make(~userName, ~offset, ~limit=Config.sketchListLimit, ());
 
+let updateQuery = (prev, next) => {
+  /* Update Apollo Store with [@bs.raw {||}] for now, since the type comming in is a generic Js.Json.t for now*/
+
+  let result = [%raw
+    {|
+        function (prev, next) {
+
+
+          var fetchMoreResult = next.fetchMoreResult;
+          console.log(Array.isArray(prev.note));
+          if (!fetchMoreResult) return prev;
+          var foundNotes = [];
+          var note = [].concat(prev.note, fetchMoreResult.note).filter(function (n) {
+            var isPresent = foundNotes.indexOf(n.id) !== -1;
+            if (isPresent) {
+              return false;
+            }
+            foundNotes.push(n.id);
+            return true;
+          });
+          return Object.assign({}, prev, {
+            note: note
+          });
+        }
+       |}
+  ];
+  result(prev, next);
+};
+
+let shouldFetchMore = (note, count) =>
+  Array.length(note) >= Config.sketchListLimit * count;
+
+type state = {count: int};
+type action =
+  | ChangeCount;
+
+let initialState = () => {count: 1};
+
+let reducer = (action, state) =>
+  switch (action) {
+  | ChangeCount => ReasonReact.Update({count: 1 + state.count})
+  };
+
+let component = ReasonReact.reducerComponent("User");
 let make = (~userName, _children) => {
   ...component,
-  render: _self => {
-    let notesQuery = GetNotes.make(~userName, ());
+  initialState,
+  reducer,
+  render: self => {
+    let notesQuery = getNotesQuery(~userName, ());
     <section className="Layout__center User">
       <h1> {j|$(userName)'s sketches|j}->str </h1>
       <GetNotesComponent
         fetchPolicy="network-only" variables=notesQuery##variables>
         ...{
-             ({result}) =>
+             ({result, fetchMore}) =>
                switch (result) {
                | Loading =>
                  <div style={ReactDOMRe.Style.make(~width="500px", ())}>
                    <UI_SketchList.Placeholder />
                  </div>
                | Error(error) => error##message->str
                | Data(response) =>
-                 <UI_SketchList
-                   sketches=response##note
-                   noSketches={<UI_NoSketches />}
-                 />
+                 let notesQuery =
+                   getNotesQuery(
+                     ~userName,
+                     ~offset=Array.length(response##note),
+                     (),
+                   );
+                 let fetchMore = _e => {
+                   fetchMore(
+                     ~variables=notesQuery##variables,
+                     ~updateQuery,
+                     (),
+                   )
+                   |> ignore;
+                   self.send(ChangeCount);
```

Comment:
`self.send` should happen inside the promise chain. The request could fail as well.